public/utils/helpers.ts

```diff
@@ -45,3 +49,101 @@ export function diffJson(oldJson?: Record<string, any>, newJson?: Record<string,
   }
   return initial + addOrChanged + oldKeys.length;
 }
+
+// code related to filter blocked index/alias/datastream
+// an example to use:
+// import { aliasBlockedPredicate, filterBlockedItems } from "./helpers";
+// import { IndexOpBlocksType } from "./constants";
+// const result = filterBlockedItems<IAlias>(services, selectedItems, IndexOpBlocksType.Closed, aliasBlockedPredicate)
+
+interface BlockedIndices {
+  [indexName: string]: String[];
+}
+
+export async function getBlockedIndices(broserServices: BrowserServices): Promise<BlockedIndices> {
+  interface ClusterBlocksStateResponse {
+    blocks: {
+      indices?: {
+        [indexName: string]: {
+          [blockId: string]: {
+            description: string;
+            retryable: boolean;
+            levels: string[];
+          };
+        };
+      };
+    };
+  }
+  const result = await broserServices.commonService.apiCaller<ClusterBlocksStateResponse>({
+    endpoint: "cluster.state",
+    data: {
+      metric: "blocks",
+    },
+  });
+  if (!result.ok) {
+    throw result.error;
+  }
+
+  const blocksResponse = result.response.blocks;
+  if (!blocksResponse.indices) {
+    return {};
+  }
+
+  const blockedIndices: BlockedIndices = {};
+  Object.keys(blocksResponse.indices).forEach((indexName) => {
+    const innerBlocksInfo = blocksResponse.indices![indexName];
+    const indexOpBlocksIds = Object.keys(innerBlocksInfo);
+    blockedIndices[indexName] = indexOpBlocksIds;
+  });
+  return blockedIndices;
+}
+
+export interface FilteredBlockedItems<T> {
+  unBlockedItems: T[];
+  blockedItems: T[];
+}
+
+export function indexBlockedPredicate(item: CatIndex, blockedItemsSet: Set<string>): boolean {
```

Comment:
A parameters with CatIndex is too much for this function, it only consume with the index params here but requires a big set of other useless parameters. The same as IAlias and DataStream, I would recommend using the type: Pick<CatIndex, "index">.